File: src/kns_indexer/models/username.py
from datetime import datetime
from typing import final

from sqlalchemy import TIMESTAMP
from sqlalchemy.orm import Mapped, mapped_column

from kns_indexer.models.base import Base


@final
class UsernameModel(Base):
    username: Mapped[str] = mapped_column(primary_key=True)
    address: Mapped[str]
    owner: Mapped[str]
    cid: Mapped[str | None]
    timestamp: Mapped[datetime] = mapped_column(
        TIMESTAMP(timezone=True),
    )

File: models/username.go
package models

import "time"

type Username struct {
	Username  string    `json:"username" example:"username" db:"username"`
	Address   string    `json:"address" example:"keeta_aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" db:"address"`
	Owner     string    `json:"owner" example:"keeta_bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb" db:"owner"`
	CID       *string   `json:"cid,omitempty" example:"Qmaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" db:"cid"`
	IsPrimary bool      `json:"isPrimary" example:"false" db:"is_primary"`
	Timestamp time.Time `json:"timestamp" example:"2025-11-25T11:22:33.123Z" db:"timestamp"`
}
